framework/framework.go

```diff
@@ -67,7 +147,7 @@ type Framework struct {
 //         AddFunc:    addFunc,
 //     }
 //
-func (f *Framework) ProcessCreate(obj interface{}, resources []Resource) error {
+func ProcessCreate(obj interface{}, resources []Resource) error {
```

Comment:
These methods are now decoupled from the framework because they do not obtain any state. They only implement logic. This makes the usage of the framework possible in two ways. Users can either use the cache resource event handler or implement add/delete/update function on their own by using the `Process*` methods explicitly.

Reply:
Yes I like the flexibility here.